surfactant/infoextractors/file_decompression.py

```diff
@@ -33,11 +33,20 @@
 RAR_SUPPORT = {"enabled": True}
 
 
-def supports_file(filetype: str) -> Optional[str]:
-    if filetype in {"TAR", "GZIP", "ZIP", "BZIP2", "XZ"} or (
-        filetype == "RAR" and RAR_SUPPORT["enabled"]
-    ):
-        return filetype
+def supports_file(filetype: list[str]) -> Optional[list[str]]:
+    if filetype is None:
+        return None
+    supported_types = {"TAR", "GZIP", "ZIP", "BZIP2", "XZ"}
+    supported = []
+    # Filter out non-archive types
+    for ft in filetype:
+        if ft in supported_types:
+            supported.append(ft)
+        elif ft == "RAR" and RAR_SUPPORT["enabled"]:
+            supported.append(ft)
```

Comment:
```suggestion
    if ft == "RAR" and RAR_SUPPORT["enabled"]:
        supported.append(ft)
```
This check can be done after checking the other filetypes has been done (only does the check once instead of on every loop iteration).